store/aerospike/aerospike.go

```diff
@@ -0,0 +1,322 @@
+package aerospike
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/abronan/valkeyrie"
+	"github.com/abronan/valkeyrie/store"
+	api "github.com/aerospike/aerospike-client-go"
+	"github.com/aerospike/aerospike-client-go/types"
+)
+
+// Aerospike is the receiver type for the
+// Store interface
+type Aerospike struct {
+	sync.Mutex
+	client    *api.Client
+	namespace string
+}
+
+var (
+	// ErrAerospikeNamespaceMissing is thrown when Namespace config option is missing
+	ErrAerospikeNamespaceMissing = errors.New("Namespace (Bucket) config option missing")
+)
+
+// Register registers aerospike to valkeyrie
+func Register() {
+	valkeyrie.AddStore(store.AEROSPIKE, New)
+}
+
+// New creates a new Aerospike client given a list
+// of endpoints and optional tls config
+func New(endpoints []string, options *store.Config) (store.Store, error) {
+
+	s := &Aerospike{}
+
+	hosts := []*api.Host{}
+	for _, endpoint := range endpoints {
+		h := strings.Split(endpoint, ":")
+		p, err := strconv.Atoi(h[1])
```

Comment:
We should wrap the access to the slice by adding a condition here:

```go
if len(h) >= 2 {
    p, err := strconv.Atoi(h[1])
    if err != nil {
 	return nil, err
    }
    hosts = append(hosts, api.NewHost(h[0], p))
}
```

Otherwise this could panic if the slice has 0 or 1 elements after the Split. (Input address is wrong for example)

Reply:
ok